### PR TITLE
Working share-engine for share extension

### PR DESCRIPTION
**Pull request by Einh06**
**What's in this PR**
This PR has a working implementation of the share engine used for the share extension.

- [x] Release of `wire-ios-message-strategy`

### CODE REVIEW

File: wire-ios-share-engine/SharingSession.swift [hunk 1]
```diff
@@ -75,74 +149,125 @@ public class SharingSession {
...
+        let fileAssetcache = FileAssetCache(location: cachesURL)
+        userInterfaceContext.zm_fileAssetCache = fileAssetcache
+        syncContext.zm_fileAssetCache = fileAssetcache
     }
 
     /// Cancel all pending tasks.
     /// Should be called when the extension is dismissed
-    func cancelAllPendingTasks() {
+    public func cancelAllPendingTasks() {
         // TODO
```
Comment (daehn) [inline wire-ios-share-engine/SharingSession.swift]: Should we fetch all not sent messages here and cancel them or is this done somewhere else?

Comment (Einh06) [inline wire-ios-share-engine/SharingSession.swift]: This is actually done in the UI so far. This is a good suggestion though and will look into it for future work.

**Comment by daehn** [comment]
@Einh06 I merged and released `wire-ios-message-strategy` version 11.0.0

File: wire-ios-share-engineTests/OperationLoopTests.swift [hunk 2]
```diff
@@ -0,0 +1,150 @@
+//
+//  OperationLoopTests.swift
```
Comment (daehn) [inline wire-ios-share-engineTests/OperationLoopTests.swift]: Missing copyright header

File: wire-ios-share-engineTests/OperationLoopTests.swift [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        if uiMoc != nil {
+            uiMoc.globalManagedObjectContextObserver.tearDown()
+        }
+        
+        if syncMoc != nil {
+            syncMoc.performGroupedBlock {
+                self.syncMoc.globalManagedObjectContextObserver.tearDown()
+                self.syncMoc.userInfo.removeAllObjects()
+            }
+            _ = waitForAllGroupsToBeEmpty(withTimeout: 0.5)
```
Comment (daehn) [inline wire-ios-share-engineTests/OperationLoopTests.swift]: Could be wrapped in a `XCTAssert()`

File: wire-ios-share-engineTests/OperationLoopTests.swift [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        XCTAssertTrue(self.waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+        
+        XCTAssertNotNil(syncUser)
+        XCTAssertNil(syncUser.name)
+        
+        var uiUser : ZMUser! = nil
+        uiMoc.performGroupedBlock {
+            uiUser = ZMUser(remoteID: userID, createIfNeeded: false, in: self.uiMoc)!
+        }
+        XCTAssertTrue(self.waitForAllGroupsToBeEmpty(withTimeout: 0.5))
```
Comment (daehn) [inline wire-ios-share-engineTests/OperationLoopTests.swift]: I would add an assert that `uiUser` is non-nil, if the user is available on the ui context we already showed that changes (in this case an insertion) are merged to the sync context.

File: wire-ios-share-engineTests/RequestGeneratorStoreTests.swift [hunk 5]
```diff
@@ -0,0 +1,147 @@
+//
+//  RequestGeneratorStoreTests.swift
```
Comment (daehn) [inline wire-ios-share-engineTests/RequestGeneratorStoreTests.swift]: Missing copyright headers

File: wire-ios-share-engineTests/RequestGeneratorStoreTests.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        internal func nextRequest() -> ZMTransportRequest? {
+            return requestBlock()
+        }
+    }
+    
+    
+    var mockStrategy = MockStrategy()
+    var sut : RequestGeneratorStore! = nil
+    
+    override func setUp() {
```
Comment (daehn) [inline wire-ios-share-engineTests/RequestGeneratorStoreTests.swift]: Why are we overriding this? Maybe we should add a comment.

Comment (daehn) [inline wire-ios-share-engineTests/RequestGeneratorStoreTests.swift]: Same for the teardown.

**Comment by marcoconti83** [comment]
Is this still "DO NOT MERGE"?

**Comment by marcoconti83** [comment]
Fixed all comments

**Pull request merged**